src/dashboard/src/components/administration/views.py

```diff
@@ -223,6 +229,160 @@ def storage(request):
+
+    return render(request, 'administration/usage.html', locals())
+
+def _usage_dirs(calculate_usage=True):
+    # Put spaces before directories contained by the spaces
+    #
+    # Description is optional, but either a path or a location purpose (used to
+    # look up the path) should be specified
+    #
+    # If only certain sudirectories within a path should be deleted, set
+    # 'subdirectories' to a list of them
+    dir_defs = (
+        ('shared', {
+            'path': _usage_shared_dir_path(),
+        }),
+        ('dips', {
+            'description': 'DIP uploads',
+            'path': os.path.join('watchedDirectories', 'uploadedDIPs'),
+            'contained_by': 'shared'
+        }),
+        ('rejected', {
+            'description': 'Rejected',
+            'path': 'rejected',
+            'contained_by': 'shared'
+        }),
+        ('failed', {
+            'description': 'Failed',
+            'path': 'failed',
+            'contained_by': 'shared'
+        }),
+        ('tmp', {
+            'description': 'Temporary file storage',
+            'path': 'tmp',
+            'contained_by': 'shared'
+        })
+    )
+
+    dirs = collections.OrderedDict(dir_defs)
+
+    # Resolve location paths and make relative paths absolute
+    for directory in dirs:
+        if dirs[directory].get('location_purpose') is not None:
+            # If a location, determine path
+            dirs[directory]['path'] = _usage_location_path(dirs[directory]['location_purpose'])
+        elif dirs[directory].get('contained_by') is not None:
+            # If contained, make path absolute
+            space = dirs[directory]['contained_by']
+            absolute_path = os.path.join(dirs[space]['path'], dirs[directory]['path'])
+            dirs[directory]['path'] = absolute_path
+
+    # Calculate usage/size, if requested
+    if calculate_usage:
+        for directory in dirs:
+            if dirs[directory].get('contained_by') is None:
+               # Get size/usage of space
+                space_path = dirs[directory]['path']
+                dirs[directory]['size'] = _usage_check_directory_volume_size(space_path)
+                dirs[directory]['used'] = _usage_check_directory_used(space_path)
+            else:
+                # Get size from containing space and calculate usage of specific path
+                space = dirs[directory]['contained_by']
+                dirs[directory]['size'] = dirs[space]['size']
+                dirs[directory]['used'] = _usage_check_directory_used(dirs[directory]['path'])
+
+    return dirs
+
+def _usage_check_directory_volume_size(path):
+    # Get volume size (in 512 byte blocks)
+    try:
+        output = subprocess.check_output(["df", path])
+        device, size, used, available, percent, mountpoint = output.split("\n")[1].split()
+        return int(size) * 512
+    except subprocess.CalledProcessError:
+        return 0
+
+def _usage_check_directory_used(path):
```

Comment:
Python convention uses [docstrings](https://www.python.org/dev/peps/pep-0257/) to summarize what a function does at a high level, and comments for descriptions inside a function.  Since this looks like a summary, should be triple quoted.